sui/p2p_transfer/index.js

```diff
@@ -19,48 +20,47 @@ const main = async () => {
   const SENDER_PRIVATE_KEY = process.env.ACC1_PRIVATE_KEY;
   const sender_keypair = getKeyPairFromExportedPrivateKey(SENDER_PRIVATE_KEY);
 
-  const RECIEVER_PRIVATE_KEY = process.env.ACC2_PRIVATE_KEY;
-  const receiver_keypair = getKeyPairFromExportedPrivateKey(RECIEVER_PRIVATE_KEY);
-  const receiver_address = receiver_keypair.getPublicKey().toSuiAddress();
-
 
   // create a new SuiClient object pointing to the network you want to use
   let url = getFullnodeUrl('mainnet');
   if (URL_OVERRIDE) {
       url = URL_OVERRIDE;
   }
   const suiClient = new SuiClient({ url: url });
+  let gasCoin = null
 
   while (true) {
     try {
       const txb = new TransactionBlock();
       const [coin] = txb.splitCoins(txb.gas, [txb.pure(1)]);
-      txb.transferObjects([coin], receiver_address);
+      txb.transferObjects([coin], sender_keypair.toSuiAddress());
+      txb.setSender(sender_keypair.toSuiAddress());
+      txb.setGasBudget(5_000_000)
+
+      // This doesn't change e2e latency, but is recommended for saving an extra rpc call during the build phase
+      if (gasCoin) {
+        txb.setGasPayment([gasCoin]);
+      }
+
+      const buildStartTime = performance.now();
+      const bytes = await txb.build({ client: suiClient });
```

Comment:
Also, what is happening in build phase ?

We notice this takes around 400ms, which seems a little on the higher side

Reply:
On the first run it should be fetching coins owned by the address to use for gas, but this should be a lot faster than what you are seeing.  For the shared version it will get the initial_shared_version of the input object, and determine if is being used mutably, but for this version it should only be that one request, so I'm not sure why you are seeing something so slow.  What rpc node are you using?

Reply:
When I run this, it's taking 45-70ms for the build process

Reply:
Where are you running this from ? We are going running it from Asia (the benchmark is run here). We are consistently seeing 400ms+..

This is not a blocker for us

Reply:
I've updated the code a bit to remove extra round trips during the build phase, which should eliminate this.  Depending on what RPC provider you are hitting, requests from asia could definitely account for a difference of a few hundred milliseconds.  I was running this from the US against RPC nodes in the US, so my round trip time is likely much lower

Reply:
I've been told that `https://mainnet.suiet.app/` is in asia, so you could try that to see how it compares